tests/test_unknown_engine.py

```diff
@@ -0,0 +1,50 @@
+import unittest
+
+from unittest import TestCase
+from testing.postgresql import Postgresql
+from pepys_import.core.store.data_store import DataStore
+from pepys_import.core.store.db_base import BaseSpatiaLite
+
+
+class UnknownDBTestCase(TestCase):
+    @unittest.expectedFailure
+    def test_unknown_database_type(self):
+        """Test whether DataStore raises exception when unknown database name entered"""
+        DataStore("", "", "", 0, "", db_type="TestDB")
+
+
+# Not working yet, metadata must be broken to raise Operational Error
+class DBConnectionTestCase(TestCase):
+    @unittest.expectedFailure
+    def test_unknown_engine_for_spatialite(self):
+        """Test whether DataStore raises exception when unknown database name entered"""
+        store = DataStore("", "", "", 0, ":memory:", db_type="sqlite")
+        BaseSpatiaLite.metadata.create_all(store.engine)
+        store.engine = None
+        # it must raise exception during initialise()
+        store.initialise()
+
+    @unittest.expectedFailure
```

Comment:
Hmm, @BarisSari  - is it possible for this test to be more specific in how the failure is caught?  If PostGIS isn't running, I guess this test will fail - but that's not because we used an invalid database name.

Reply:
Actually, this test is wrong right now. I'm sorry that I forgot to change it. I created it to hit OperationalError exception during the creation of tables. I'm going to change it or remove it. It doesn't test anything at all right now.